File: Chapter07/Java/LoggedInUserSet.java
package main.java;

import java.util.HashSet;

/**
 * Created by William Smith on 9/11/16.
 */
public class LoggedInUserSet {
    HashSet<User> _users;

    public LoggedInUserSet()
    {
        _users = new HashSet<User>();
    }

    public boolean userAuthenticated(User user)
    {
        if (_users.size() < 30)
        {
            return _users.add(user);
        }
        return false;
    }

    public void userLoggedOut(User user)
    {
        _users.remove(user);
    }
}

File: Chapter07/Swift/LoggedInUserSet.swift
//
//  LoggedInUserSet.swift
//  EverydayDataStructures-Swift
//
//  Created by William Smith on 9/11/16.
//  Copyright © 2016 Websmiths, LLC. All rights reserved.
//

import Foundation

open class LoggedInUserSet
{
    var _users: Set<User> = Set<User>()

    open func userAuthenticated(_ user: User)
    {
        if (_users.count < 30)
        {
            _users.insert(user)
        }
    }

    open func userLoggedOut(_ user: User)
    {
        if let index = _users.index(of: user)
        {
            _users.remove(at: index)
        }
    }
}
